MediaBrowser.Providers/TV/TheTVDB/TvdbSeriesProvider.cs

```diff
@@ -109,428 +92,93 @@ public async Task<MetadataResult<Series>> GetMetadata(SeriesInfo itemId, Cancell
-                                            return FindSeriesId(subtree);
-                                        }
-                                    }
-
-                                default:
-                                    reader.Skip();
-                                    break;
-                            }
-                        }
-                        else
-                        {
-                            reader.Read();
-                        }
-                    }
-                }
-            }
-
-            return null;
-        }
-
-        private string FindSeriesId(XmlReader reader)
-        {
-            reader.MoveToContent();
-            reader.Read();
-
-            // Loop through each element
-            while (!reader.EOF && reader.ReadState == ReadState.Interactive)
+            try
             {
-                if (reader.NodeType == XmlNodeType.Element)
+                if (string.Equals(idType, MetadataProviders.Zap2It.ToString(), StringComparison.OrdinalIgnoreCase))
                 {
-                    switch (reader.Name)
-                    {
-                        case "seriesid":
-                            {
-                                var val = reader.ReadElementContentAsString();
-
-                                if (!string.IsNullOrWhiteSpace(val))
-                                {
-                                    return val;
-                                }
-
-                                return null;
-                            }
-
-                        default:
-                            reader.Skip();
-                            break;
-                    }
+                    result = await _tvDbClientManager.GetSeriesByZap2ItIdAsync(id, cancellationToken);
                 }
                 else
                 {
-                    reader.Read();
-                }
-            }
-
-            return null;
-        }
-
-        internal static bool IsValidSeries(Dictionary<string, string> seriesProviderIds)
-        {
-            if (seriesProviderIds.TryGetValue(MetadataProviders.Tvdb.ToString(), out string id))
-            {
-                // This check should ideally never be necessary but we're seeing some cases of this and haven't tracked them down yet.
-                if (!string.IsNullOrWhiteSpace(id))
-                {
-                    return true;
+                    result = await _tvDbClientManager.GetSeriesByImdbIdAsync(id, cancellationToken);
                 }
             }
-
-            if (seriesProviderIds.TryGetValue(MetadataProviders.Imdb.ToString(), out id))
+            catch (TvDbServerException e)
             {
-                // This check should ideally never be necessary but we're seeing some cases of this and haven't tracked them down yet.
-                if (!string.IsNullOrWhiteSpace(id))
-                {
-                    return true;
-                }
```

Comment:
hmm, what about non-English languages?.. :)

Reply:
Which line is it?

Reply:
about changing "&" to "and" and then articles to spaces

Reply:
Dunno. Not sure there is an ideal catch-all solution to this. Again, I didn't write this, so I can't really say what inspired the function in the first place. I'm guessing they had problems matching names in Emby to names in tvdb.